lib/matplotlib/backend_bases.py

```diff
@@ -3640,7 +3683,10 @@ def trigger_tool(self, name):
             Name(id) of the tool triggered from within the toolbar
 
         """
-        self.navigation.tool_trigger_event(name, sender=self)
+        try:
+            self.navigation.tool_trigger_event(name, sender=self)
+        except:
+            self.toggle_toolitem(name)
```

Comment:
It doesn't matter where the exception comes from, if the tool_trigger_event fails, it fails.
Here you should make sure the button corresponds to a toggle.